src/main/java/com/cloudbees/plugins/credentials/SystemCredentialsProvider.java

```diff
@@ -323,6 +323,14 @@ private synchronized boolean removeCredentials(@NonNull Domain domain, @NonNull
     private synchronized boolean updateCredentials(@NonNull Domain domain, @NonNull Credentials current,
                                                    @NonNull Credentials replacement) throws IOException {
         checkPermission(CredentialsProvider.UPDATE);
+
+        // See IdCredentials.Helper#equals.
+        // Note: We probably should not assume that all credentials are IdCredentials. We also do this a few lines below in list.indexOf(current) call.
+        // If this one breaks, that one breaks as well.
```

Comment:
how so - the list is of type `Credentials` not `IdCredentials` so that would not break with a big bad angry jenkins.
At worst it would not find `current`, as the list would use `equals` and it is up to current credential may have been correctly obtained by it's name or otherwise in the code path.